lab1/main.cpp

```diff
@@ -0,0 +1,201 @@
+    while (true) {
+        switch (menu()) {
+            case 1:
+                readFromFile(FILE_NAME);
+                break;
+            case 2:
+                saveToFile(FILE_NAME);
+                break;
+            case 3:
+                addNew();
+                break;
+            case 4:
+                del();
+                break;
+            case 5:
+                sort_data();
+                break;
+            case 6:
+                modify();
+                break;
+            case 7:
+                return 0;
+            default:
+                cout << "Wrong choice!" << endl;
+        }
+    }
+}
+
+int menu() {
+    cout << "\n";
+    cout << "Choose\n";
+    cout << "1-For look up from a file\n";
+    cout << "2-To write to a file\n";
+    cout << "3-To add a record\n";
+    cout << "4-To delete the record\n";
+    cout << "5-To sort by surname\n";
+    cout << "6-To change record\n";
+    cout << "7-To exit\n";
+    cout << "\n";
+    cout << "Your choice: ";
+
+    char ans_char[MAXLEN];
+    int ans;
+    cin.getline(ans_char, MAXLEN);
+    ans = stoi(ans_char);
+    return ans;
+}
+
+void saveToFile(const char *fileName) {
+    ofstream f;
+    f.open(fileName, ios::binary);
+    f.write((char *) arr, sizeof(Student) * student_index);
+    f.close();
+    cout << "The entered data is saved to a file\n";
+}
+
+void readFromFile(const char *fileName) {
+    ifstream f;
+    f.open(fileName, ios::binary);
+
+    if (!f) {
+        cout << "The file does not exist";
+    } else {
+        Student stud;
+        student_index = 0;
+        while (true) {
+            f.read((char *) &stud, sizeof(stud));
+            if (f.eof())
+                break;
+            arr[student_index] = stud;
+            student_index++;
+        }
+        f.close();
+        cout << "Data read from file\n";
+        print_students();
+    }
+}
+
+void addNew() {
+    cout << "Adding a new record\n\n";
+    cout << "Record number " << student_index + 1 << "\n";
+    cout << "Enter your surname: ";
+    cin.getline(arr[student_index].surname, MAXLEN);
+    cout << "Enter the group number: ";
+    cin.getline(arr[student_index].group_number, MAXLEN);
+    cout << "Enter the address: ";
+    cin.getline(arr[student_index].address, MAXLEN);
+    cout << "Enter a phone number: ";
+    cin.getline(arr[student_index].phone_number, MAX_NUM);
+
+    student_index++;
+    cout << "\n";
+    print_students();
+}
+
+void del() {
+    char student_number_str[MAXLEN];
+    int student_number;
+
+    cout << "Select the record number to be deleted: ";
+    cin.getline(student_number_str, MAXLEN);
+    student_number = stoi(student_number_str);
+
+    if (student_number < 0 || student_number > student_index) {
+        cout << "Wrong index!";
+        return;
+    }
+
+    for (int i = student_number - 1; i < student_index; i++) {
+        arr[i] = arr[i + 1];
+    }
+    student_index--;
+
+    cout << "\n";
+    print_students();
```

Comment:
if/else -- don't use empty return
move the second part to a separate function

Reply:
I didn't understand what to do